### PR TITLE
Allow on-the-fly reassign when workplace is scanned

### PR DESCRIPTION
**Pull request by metas-ts**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java [hunk 1]
```diff
@@ -50,7 +50,27 @@ public Optional<WorkplaceId> getWorkplaceIdByUserId(@NonNull final UserId userId
...
+		syncAndSave(request, record);
+	}
+
+	public void createOrUpdate(@NonNull final WorkplaceAssignmentCreateRequest request)
+	{
+		final Optional<WorkplaceId> workplaceId = getWorkplaceIdByUserId(request.getUserId());
+		final I_C_Workplace_User_Assign record;
+		if (workplaceId.isPresent())
+		{
+			record = InterfaceWrapperHelper.load(workplaceId.get(), I_C_Workplace_User_Assign.class);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java:62]: this looks weird to me....

Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java:62]: I think u should search by workspace id and user id!

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java (hunk 1 above)
Comment (metas-ts) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java:62]: ofc, right. thx!

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java [hunk 2]
```diff
@@ -50,7 +50,24 @@ public Optional<WorkplaceId> getWorkplaceIdByUserId(@NonNull final UserId userId
...
+				.addEqualsFilter(I_C_Workplace_User_Assign.COLUMNNAME_AD_User_ID, request.getUserId())
+				.orderByDescending(I_C_Workplace_User_Assign.COLUMNNAME_AD_Org_ID) // TODO have org-id in the request as we have a UC regarding it
+				.create()
+				.firstOptional(I_C_Workplace_User_Assign.class)
+				.orElseGet(() -> InterfaceWrapperHelper.newInstance(I_C_Workplace_User_Assign.class));
+
+		syncAndSave(request, record);
+	}
 
+	private static void syncAndSave(final WorkplaceAssignmentCreateRequest request, final I_C_Workplace_User_Assign record)
```
[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java:69]: extract InterfaceWrapperHelper.save from here.

[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java:69]: text search code for `updateRecord` to see how we approach it

**Comment by teosarca** [comment]
closing since it's legacy

**Pull request closed without merging**